haptools/sim_genotype.py

```diff
@@ -832,19 +904,27 @@ def validate_params(model, mapdir, chroms, popsize, invcf, sample_info, region=N
 
     # validate sample_info file (ensure pops given are in model file and samples in vcf file)
     # ensure sample_info col 2 in pops
+    total_per_pop = {}
     sample_pops = set()
     for line in open(sample_info, 'r'):
         sample = line.split()[0]
         info_pop = line.split()[1]
         sample_pops.add(info_pop)
+        if not total_per_pop.get(info_pop, False):
+            total_per_pop[info_pop] = 1
+        else:
+            total_per_pop[info_pop] += 1
 
         if sample not in vcf_samples:
             raise Exception(f"Sample {sample} in sampleinfo file is not present in the vcf file.")
     
-    # Ensure that all populations from the model file are listed in the sample info file
+    # Ensure that all populations from the model file are listed in the sample info file 
+    #    and that there is sufficient enough samples when no_replacement is specified
     for model_pop in pops[1:]:
         if model_pop not in list(sample_pops):
             raise Exception(f"Population {model_pop} in model file is not present in the sample info file.")
+        if no_replacement and total_per_pop[model_pop] < num_samples:
+            raise Exception(f"Population {model_pop} does not have enough samples to sample without replacement. Please ensure that each population specified has >= total number of samples output: {num_samples}")
```

Comment:
Technically, it is possible to simulate with `< num_samples` per population. For example, if we simulated data with 50% ancestry A and 50% ancestry B, we could often get by with only `.6 * num_samples` individuals per population.

Reply:
I agree, but if we simulate a large number of samples/generations it would be unfortunate to have the process exit due to an error of not having enough samples to simulate, so I think it's best we make sure there's sufficient samples to simulate from.